From c6ae355b8f18f40a0642edfa456cc0d3c006bfed Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 065/296] Widget: Catch null point exception in
 AbsListViewAutoScroller

The first view of AbsListView is null, so there is a NullPointerException
happen.

Add a judgement for the first view of AbsListView.
---
 core/java/com/android/internal/widget/AutoScrollHelper.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/java/com/android/internal/widget/AutoScrollHelper.java b/core/java/com/android/internal/widget/AutoScrollHelper.java
index 0d468ca2494..ee4f28dfafc 100644
--- a/core/java/com/android/internal/widget/AutoScrollHelper.java
+++ b/core/java/com/android/internal/widget/AutoScrollHelper.java
@@ -912,7 +912,7 @@ public abstract class AutoScrollHelper implements View.OnTouchListener {
                 // Are we already showing the entire first item?
                 if (firstPosition <= 0) {
                     final View firstView = target.getChildAt(0);
-                    if (firstView.getTop() >= 0) {
+                    if (firstView == null || firstView.getTop() >= 0) {
                         return false;
                     }
                 }
-- 
2.11.0

